src/test/java/io/lettuce/core/cluster/pubsub/RedisClusterPubSubConnectionIntegrationTests.java

```diff
@@ -126,13 +133,55 @@ void subscribeToShardChannelViaReplica() {
         int clusterKeyslot = connection.sync().clusterKeyslot(shardChannel).intValue();
         String thisNode = connection.getPartitions().getPartitionBySlot(clusterKeyslot).getNodeId();
 
-        RedisPubSubAsyncCommands<String, String> replica =
-                pubSubConnection.async().nodes(node -> thisNode.equals(node.getSlaveOf())).commands(0);
+        RedisPubSubAsyncCommands<String, String> replica = pubSubConnection.async()
+                .nodes(node -> thisNode.equals(node.getSlaveOf())).commands(0);
         replica.ssubscribe(shardChannel);
 
         Wait.untilEquals(shardChannel, connectionListener.getShardChannels()::poll).waitOrTimeout();
     }
 
+    @Test
+    @EnabledOnCommand("SSUBSCRIBE")
+    void publishToShardChannel() throws Exception {
+        pubSubConnection.addListener(connectionListener);
+        pubSubConnection.async().ssubscribe(shardChannel);
+        Wait.untilEquals(shardChannel, connectionListener.getShardChannels()::poll).waitOrTimeout();
+
+        pubSubConnection.async().spublish(shardChannel, shardMessage);
+        Wait.untilEquals(shardChannel, connectionListener.getShardChannels()::poll).waitOrTimeout();
+        Wait.untilEquals(shardMessage, connectionListener.getMessages()::poll).waitOrTimeout();
+    }
+
+    @Test
+    @EnabledOnCommand("SSUBSCRIBE")
+    void publishToShardChannelViaDifferentEndpoints() throws Exception {
+        pubSubConnection.addListener(connectionListener);
+        pubSubConnection.async().ssubscribe(shardChannel);
+        Wait.untilEquals(shardChannel, connectionListener.getShardChannels()::poll).waitOrTimeout();
+
+        pubSubConnection.async().ssubscribe(shardTestChannel);
```

Comment:
It would make sense to have a test that operates on a single shard connection to ensure that no routing (`MOVED` redirection) happens.

Cluster connections transparently handle redirects.

Reply:
could you eloborate it ? 
are you looking for additional tests where we just use node listener instead of connection listener ?

Reply:
I just realized that we cannot easily test `MOVED` redirections unless setting `ClusterClientOptions.maxRedirects` to `0`.

That would be in any case worth testing for Pub/Sub connections to open a new connection and make sure that no redirects happen to ensure maximum performance.

In my first comment, I mixed up things so please disregard the "single shard connection" aspect.

Reply:
LMK if  [this](https://github.com/redis/lettuce/pull/2838/commits/b906da7413933545b2344f624483e72260862c5b) covers what you have in mind.